src/lib/times.js

```diff
@@ -77,7 +78,8 @@ export function confirmationTime(time, timeType) {
   if (upcoming.minutes()) {
     timeStr = getLocalizedDateTime(upcoming, 'confirmation_time');
   }
-  const dateStr = getLocalizedDateTime(upcoming, 'confirmation_date');
+  const dateStr = getLocalizedDateTime(upcoming,
+    (upcoming.year() === thisYear) ? 'confirmation_date' : 'confirmation_date_with_year');
```

Comment:
Oh, my other thought was that we could only show the date if the year wasn't this year, and the month and day was after today.  (i.e. on Dec 31st, a date of Jan 17th isn't going to be assumed to be in this year.)  But that's not really needed in this patch, and I'm not entirely sure it's a good idea.  🙂